File: src/main/java/waterpower/integration/waila/HUDHandlerReservoir.java
/**
 * Copyright (c) Huang Yuhui, 2014
 * 
 * "WaterCraft" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.integration.waila;

import java.util.List;

import mcp.mobius.waila.api.IWailaConfigHandler;
import mcp.mobius.waila.api.IWailaDataAccessor;
import mcp.mobius.waila.api.IWailaDataProvider;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.client.resources.I18n;
import net.minecraft.world.World;
import net.minecraftforge.fluids.FluidStack;
import net.minecraftforge.fml.common.Optional.Interface;
import net.minecraftforge.fml.common.Optional.InterfaceList;
import net.minecraftforge.fml.common.Optional.Method;
import waterpower.client.Local;
import waterpower.common.block.reservoir.TileEntityReservoir;
import waterpower.util.Mods;

@InterfaceList({ @Interface(iface = "mcp.mobius.waila.api.IWailaDataProvider", modid = Mods.IDs.Waila) })
public class HUDHandlerReservoir implements IWailaDataProvider {

    @Override
    @Method(modid = Mods.IDs.Waila)
    public List<String> getWailaBody(ItemStack arg0, List<String> arg1, IWailaDataAccessor arg2, IWailaConfigHandler arg3) {
        TileEntity te = arg2.getTileEntity();
        if (!(te instanceof TileEntityReservoir))
            return arg1;
        TileEntityReservoir tile = (TileEntityReservoir) te;
        arg1.add(Local.get("cptwtrml.gui.reservoir.add") + ": " + tile.getLastAddedWater());
        if (tile.getFluidTank() == null)
            return arg1;
        FluidStack f = tile.getFluidTank().getFluid();
        arg1.add((f == null ? Local.get("cptwtrml.gui.empty") : f.getLocalizedName()) + ": " + tile.getFluidAmount() + "/"
                + tile.getFluidTankCapacity() + "mb");
        return arg1;
    }

    @Override
    @Method(modid = Mods.IDs.Waila)
    public List<String> getWailaHead(ItemStack arg0, List<String> arg1, IWailaDataAccessor arg2, IWailaConfigHandler arg3) {
        return arg1;
    }

    @Override
    @Method(modid = Mods.IDs.Waila)
    public ItemStack getWailaStack(IWailaDataAccessor arg0, IWailaConfigHandler arg1) {
    	return arg0.getStack();
    }

    @Override
    @Method(modid = Mods.IDs.Waila)
    public List<String> getWailaTail(ItemStack arg0, List<String> arg1, IWailaDataAccessor arg2, IWailaConfigHandler arg3) {
        return arg1;
    }

    @Override
    @Method(modid = Mods.IDs.Waila)
    public NBTTagCompound getNBTData(EntityPlayerMP arg0, TileEntity arg1, NBTTagCompound arg2, World arg3, BlockPos pos) {
        return null;
    }

}

File: src/main/kotlin/waterpower/integration/waila/HUDHandlerReservoir.kt
/**
 * Copyright (c) Huang Yuhui, 2017
 *
 * "WaterPower" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.integration.waila


import mcp.mobius.waila.api.IWailaConfigHandler
import mcp.mobius.waila.api.IWailaDataAccessor
import mcp.mobius.waila.api.IWailaDataProvider
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.Optional.*
import waterpower.client.i18n
import waterpower.common.block.reservoir.TileEntityReservoir
import waterpower.integration.IDs
import waterpower.util.emptyStack

@InterfaceList(Interface(iface = "mcp.mobius.waila.api.IWailaDataProvider", modid = IDs.Waila))
object HUDHandlerReservoir : IWailaDataProvider {

    @Method(modid = IDs.Waila)
    override fun getWailaBody(stack: ItemStack, tips: MutableList<String>, accessor: IWailaDataAccessor, configHandler: IWailaConfigHandler): List<String> {
        val te = accessor.tileEntity as? TileEntityReservoir ?: return tips
        tips.add(i18n("waterpower.gui.reservoir.add") + ": " + te.getLastAddedWater())
        val fluidTank = te.getFluidTank()
        tips.add((fluidTank.fluid?.localizedName ?: i18n("waterpower.gui.empty")) + ": " + fluidTank.fluidAmount + "mb")
        tips.add(i18n("waterpower.gui.capacity") + ": " + fluidTank.capacity + "mb")
        return tips
    }

    @Method(modid = IDs.Waila)
    override fun getWailaHead(stack: ItemStack, tips: List<String>, accessor: IWailaDataAccessor, configHandler: IWailaConfigHandler): List<String> {
        return tips
    }

    @Method(modid = IDs.Waila)
    override fun getWailaStack(accessor: IWailaDataAccessor, configHandler: IWailaConfigHandler): ItemStack? {
        return emptyStack
    }

    @Method(modid = IDs.Waila)
    override fun getWailaTail(stack: ItemStack, tips: List<String>, accessor: IWailaDataAccessor, configHandler: IWailaConfigHandler): List<String> {
        return tips
    }

    @Method(modid = IDs.Waila)
    override fun getNBTData(player: EntityPlayerMP, te: TileEntity, tag: NBTTagCompound, world: World, pos: BlockPos): NBTTagCompound {
        return tag
    }

}